internal/apptoken/app_token_usage_test.go

```diff
@@ -0,0 +1,182 @@
+		testRequestMethod    = "GET"
+		testRequestPath      = "/test-resource"
+	)
+
+	tests := []struct {
+		name             string
+		appTokenId       string
+		clientTcpAddress string
+		requestMethod    string
+		requestPath      string
+		createdTime      time.Time
+		want             *AppTokenUsage
+		wantErrContains  string
+		wantErrMatch     *errors.Template
+	}{
+		{
+			name:             "success",
+			appTokenId:       testAppTokenId,
+			clientTcpAddress: testClientTcpAddress,
+			requestMethod:    testRequestMethod,
+			requestPath:      testRequestPath,
+			createdTime:      testTime,
+			want: &AppTokenUsage{
+				AppTokenUsage: &store.AppTokenUsage{
+					AppTokenId:       testAppTokenId,
+					ClientTcpAddress: testClientTcpAddress,
+					RequestMethod:    testRequestMethod,
+					RequestPath:      testRequestPath,
+					CreateTime:       &timestamp.Timestamp{Timestamp: timestamppb.New(testTime)},
+				},
+			},
+		},
+		{
+			name:             "missing-app-token-id",
+			clientTcpAddress: testClientTcpAddress,
+			requestMethod:    testRequestMethod,
+			requestPath:      testRequestPath,
+			createdTime:      testTime,
+			wantErrContains:  "missing  app token id",
+			wantErrMatch:     errors.T(errors.InvalidParameter),
+		},
+		{
+			name:            "missing-client-tcp-address",
+			appTokenId:      testAppTokenId,
+			requestMethod:   testRequestMethod,
+			requestPath:     testRequestPath,
+			createdTime:     testTime,
+			wantErrContains: "missing created by cleint tcp address",
+			wantErrMatch:    errors.T(errors.InvalidParameter),
+		},
+		{
+			name:             "missing-created-time",
+			appTokenId:       testAppTokenId,
+			clientTcpAddress: testClientTcpAddress,
+			requestMethod:    testRequestMethod,
+			requestPath:      testRequestPath,
+			wantErrContains:  "missing created time",
+			wantErrMatch:     errors.T(errors.InvalidParameter),
+		},
+		{
+			name:             "missing-request-method",
+			appTokenId:       testAppTokenId,
+			clientTcpAddress: testClientTcpAddress,
+			createdTime:      testTime,
+			requestPath:      testRequestPath,
+			wantErrContains:  "missing request method",
+			wantErrMatch:     errors.T(errors.InvalidParameter),
+		},
+		{
+			name:             "missing-request-path",
+			appTokenId:       testAppTokenId,
+			clientTcpAddress: testClientTcpAddress,
+			createdTime:      testTime,
+			requestMethod:    testRequestMethod,
+			wantErrContains:  "missing request path",
+			wantErrMatch:     errors.T(errors.InvalidParameter),
+		},
+	}
+	for _, tc := range tests {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			assert, require := assert.New(t), require.New(t)
+			got, err := NewAppTokenUsage(testCtx, tc.appTokenId, tc.clientTcpAddress, tc.requestMethod, tc.requestPath, tc.createdTime)
+			if tc.wantErrContains != "" {
+				require.Error(err)
+				if tc.wantErrMatch != nil {
+					assert.True(errors.Match(tc.wantErrMatch, err))
+				}
+				return
```

Comment:
perhaps assert the value of tc.wantErrContains?  I bet I missed this in my PR too.